mesonbuild/compilers/compilers.py

```diff
@@ -923,11 +923,14 @@ def get_largefile_args(compiler):
 def gnulike_default_include_dirs(compiler, lang):
     if lang == 'cpp':
         lang = 'c++'
+    env = os.environ.copy()
+    env["LC_ALL"] = 'C'
     p = subprocess.Popen(
         compiler + ['-x{}'.format(lang), '-E', '-v', '-'],
         stdin=subprocess.DEVNULL,
         stderr=subprocess.PIPE,
-        stdout=subprocess.PIPE
+        stdout=subprocess.PIPE,
+        env=env
     )
     stderr = p.stderr.read().decode('utf-8')
```

Comment:
Using `'C'` and `'utf-8'` is inconsistent, no? And `C.utf8` is not necessarily widespread. Would setting `LC_MESSAGES` work?

Reply:
Would the compiler complain though?

Reply:
If there are paths that cannot be represented with ASCII, then maybe?

Reply:
@QuLogic 
I'm not a locale/encoding expert. 
I would say that setting LC_MESSAGES might be enough while setting LC_ALL isn't wrong.
Since UTF-8 can encode any char and is ASCII compatible I don't see anything better, do you have failing test cases?

Reply:
The problem is not on the Python side, but on the compiler side as `'C'` implies ASCII. Now, I don't have any non-ASCII paths, so I don't really have a test case or any idea if the compiler would even output this correctly.

Reply:
I did a test on my laptop with this folder name 'test_£_♣_が' it works. 
I'm kind of stuck. My PR mainly ensures that user local doesn't changes gcc output. 
I really don't know what should I do about UTF-8.
For now, any user with a non english linux box may suffer this issue.

Reply:
Maybe it should set an english utf-8 locale?

Reply:
I think this should be considered when a real world problem with `C` arrives. As it is now, this PR improves the situation for anyone with a non english locale A LOT.